package/linux-firmware/Config.in

```diff
@@ -8,6 +8,35 @@ config BR2_PACKAGE_LINUX_FIRMWARE
 
 if BR2_PACKAGE_LINUX_FIRMWARE
 
+config BR2_PACKAGE_LINUX_FIRMWARE_COMPRESS
+    bool "Compress firmware files"
+    help
+      Compress firmware files to save space in the root filesystem. This
+      is not as efficient as whole root filesystem compression, but it
+      is better performance-wise, because the firmwares are usually
+      accessed only when the driver is loaded. This option requires
+      enabling the FW_LOADER_COMPRESS option in the kernel along with the
+      option specific for the compression used (FW_LOADER_COMPRESS_XZ,
+      FW_LOADER_COMPRESS_ZSTD).
```

Comment:
Buildroot allows to force enable certain configuration (using `$(call KCONFIG_ENABLE_OPT,CONFIG_POSIX_MQUEUE)`). While I am a bit skeptical of that approach at times, it is most likely required if want to upstream this.